server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -198,6 +198,52 @@ public long createOrUpdateSecret(String name, String encryptedSecret, String cre
     return ImmutableList.copyOf(results);
   }
 
+  /**
+   * @param name of secret series to look up secrets by.
+   * @param versionIdx the first index to select in a list of versions sorted by creation time
+   * @param numVersions the number of versions after versionIdx to select in the list of versions
+   * @return Versions of a secret matching input parameters or Optional.absent().
+   */
+  public Optional<SecretSeriesAndVersions> getSecretVersionsByName(String name,
+      int versionIdx,
+      int numVersions) {
+    checkArgument(!name.isEmpty());
+    checkArgument(versionIdx >= 0);
+    checkArgument(numVersions >= 0);
+
+    SecretContentDAO secretContentDAO = secretContentDAOFactory.using(dslContext.configuration());
+    SecretSeriesDAO secretSeriesDAO = secretSeriesDAOFactory.using(dslContext.configuration());
+
+    Optional<SecretSeries> series = secretSeriesDAO.getSecretSeriesByName(name);
+    if (series.isPresent()) {
+      long secretId = series.get().id();
+      Optional<ImmutableList<SecretContent>> contents =
+          secretContentDAO.getSecretVersionsBySecretId(secretId, versionIdx, numVersions);
+      if (contents.isPresent()) {
+        return Optional.of(SecretSeriesAndVersions.of(series.get(), contents.get()));
+      }
+    }
+    return Optional.empty();
+  }
+
+  /**
+   * @param name of secret series for which to reset secret version
+   * @param versionId The identifier for the desired current version
+   * @return 0 for success, non-zero for failure (secret series not found)
+   */
+  public int setCurrentSecretVersionByName(String name, long versionId) {
```

Comment:
Instead of returning 0/1, just have it throw an exception in case of failure and make the return type `void`.